cli/organization.go

```diff
@@ -20,13 +26,181 @@ func (r *RootCmd) organizations() *clibase.Cmd {
 		},
 		Children: []*clibase.Cmd{
 			r.currentOrganization(),
+			r.switchOrganization(),
 		},
 	}
 
 	cmd.Options = clibase.OptionSet{}
 	return cmd
 }
 
+func (r *RootCmd) switchOrganization() *clibase.Cmd {
+	client := new(codersdk.Client)
+
+	cmd := &clibase.Cmd{
+		Use:   "switch <organization name | ID>",
+		Short: "Switch the organization used by the cli. Pass an empty string to reset to the default organization.",
+		Long: "Switch the organization used by the cli. Pass an empty string to reset to the default organization.\n" + formatExamples(
+			example{
+				Description: "Remove the current organization and defer to the default.",
+				Command:     "coder organizations switch ''",
+			},
+			example{
+				Description: "Switch to a custom organization.",
+				Command:     "coder organizations switch my-org",
+			},
+		),
+		Middleware: clibase.Chain(
+			r.InitClient(client),
+			clibase.RequireRangeArgs(0, 1),
+		),
+		Options: clibase.OptionSet{},
+		Handler: func(inv *clibase.Invocation) error {
+			conf := r.createConfig()
+			orgs, err := client.OrganizationsByUser(inv.Context(), codersdk.Me)
+			if err != nil {
+				return fmt.Errorf("failed to get organizations: %w", err)
+			}
+			// Keep the list of orgs sorted
+			slices.SortFunc(orgs, func(a, b codersdk.Organization) int {
+				return strings.Compare(a.Name, b.Name)
+			})
+
+			var switchToOrg string
+			if len(inv.Args) == 0 {
+				// Pull switchToOrg from a prompt selector, rather than command line
+				// args.
+				switchToOrg, err = promptUserSelectOrg(inv, conf, orgs)
```

Comment:
It is! But I read it only from the config file. I ignore the `CurrentOrganization` because that takes into account the `-z` flag and `is_default`. 

If a user has nothing set, I want the reset option to be selected.